test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java

```diff
@@ -212,35 +213,21 @@ public void handle(final HttpExchange exchange) throws IOException {
                     exchange.sendResponseHeaders(RestStatus.NOT_FOUND.getStatus(), -1);
                     return;
                 }
-                byte[] blob = BytesReference.toBytes(blobs.get(blobName));
+                BytesReference blob = blobs.get(blobName);
                 final String range = exchange.getRequestHeaders().getFirst("Content-Range");
                 final Integer limit = getContentRangeLimit(range);
                 final int start = getContentRangeStart(range);
                 final int end = getContentRangeEnd(range);
 
-                final ByteArrayOutputStream out = new ByteArrayOutputStream() {
```

Comment:
Using the BAOS here was pretty brutal for multi MB requests :) Much nicer now with the `16kb` chunks `Streams.readFully(wrappedRequest)` produces